tests/baselines/reference/multipleDeclarations.types

```diff
@@ -21,7 +21,7 @@ C.prototype.m = function() {
     this.nothing();
 >this.nothing() : any
 >this.nothing : any
->this : { m: () => void; }
+>this : { m: any; }

```

Comment:
The type got worse here, I think because it unions the widened type of both declarations now. That is, it unions `any | () => void` and gets `any`. Before it must have unioned `null | () => void` which produces `() => void`.

I'm not sure there's a good way to get around this. Maybe don't widen null and undefined?